boards/FSM/Src/App/App_Signals.c

```diff
@@ -0,0 +1,52 @@
+#include "App_AcceleratorPedal.h"
+#include "App_Signals.h"
+
+struct AcceleratorPedal *_papps;
+struct AcceleratorPedal *_sapps;
+
+static bool IsPappsAlarmActive(void)
+{
+    return App_AcceleratorPedal_IsEncoderAlarmActive(_papps);
+}
+
+static void PappsAlarmCallback(void *context)
+{
+    struct FsmWorld *         world  = (struct FsmWorld *)context;
+    struct FsmCanTxInterface *can_tx = App_FsmWorld_GetCanTx(world);
+
+    App_CanTx_SetPeriodicSignal_PAPPS_MAPPED_PEDAL_PERCENTAGE(can_tx, 0.0f);
+}
+
+static bool IsSappsAlarmActive(void)
+{
+    return App_AcceleratorPedal_IsEncoderAlarmActive(_papps);
+}
+
+static void SappsAlarmCallback(void *context)
+{
+    struct FsmWorld *         world  = (struct FsmWorld *)context;
+    struct FsmCanTxInterface *can_tx = App_FsmWorld_GetCanTx(world);
+
+    App_CanTx_SetPeriodicSignal_SAPPS_MAPPED_PEDAL_PERCENTAGE(can_tx, 0.0f);
+}
+
+void App_Signals_Init(struct FsmWorld *world)
```

Comment:
This file is needed because I need to make sure the same `IsActive` and `Callback` are used for the embedded and test targets.

Reply:
If these signals are an implicit part of `FsmWorld`, why not set them up in the `FsmWorld_Create` function?

Reply:
That's a good point actually, let me try it out and  see how it looks

Reply:
I tried this out but ran into some problems with this function:
```
static bool IsPappsAlarmActive(void)                                                                                 
{                                                                                                                    
     return App_AcceleratorPedal_IsEncoderAlarmActive(_papps);                                                        
}
```

The problem is it's unclear where `_papps` should exist and when it should be initialized. I could place it in `App_FsmWorld.cpp` but it feels kind of odd.

```
// App_FsmWorld.c
static AcceleratorPedal* _papps;

static bool IsPappsAlarmActive(void)                                                                                 
{                                                                                                                    
     return App_AcceleratorPedal_IsEncoderAlarmActive(_papps);                                                        
}

FsmWorld* App_FsmWorld_Create(
  ...,
  papps)
{
  ...
  _papps = papps;
  struct Signal *papps_signal =                                                                                    
    App_SharedSignal_Create(0, IsPappsAlarmActive, papps_callback);                                              
  App_FsmWorld_RegisterSignal(world, papps_signal);  
   ...
}
```
This should work but it it feels odd because it's hardcoding the same `_papps` for every call to `App_FsmWorld_Create`. What do you think?

Reply:
Why not add an argument to `App_FsmWorld_Create`: `bool (*is_papps_alarm_active)(void)`? Then we can just pass the function in.

Reply:
Another option is to chnage `function` to take a `World`, then we can just write the function as:
```
bool IsPappsAlarmActive(FsmWorld* world) {
    return return App_AcceleratorPedal_IsEncoderAlarmActive(world->_papps);
}
```
and pass it in when we create the signal

Reply:
Went with the second option so we don't have to worry about initializing static variables `_papps` and `_sapps`. I think this new approach is quite a bit cleaner!

Reply:
Also since `RegisterSignal` is a public function, I thought it made sense to call it _outside` of `FsmWorld_Create`. Now the code calls `RegisterSignal` after `FsmWorld_Create`, i.e.

```
    world = App_FsmWorld_Create(... papps, sapps);

    struct SignalCallback papps_callback = {
        .high_duration_ms = 10,
        .function         = App_AcceleratorPedalSignals_PappsAlarmCallback,
    };
    papps_signal = App_SharedSignal_Create(
        0, App_AcceleratorPedalSignals_IsPappsAlarmActive, world,
        papps_callback);
    App_FsmWorld_RegisterSignal(world, papps_signal);

    struct SignalCallback sapps_callback = {
        .high_duration_ms = 10,
        .function         = App_AcceleratorPedalSignals_SappsAlarmCallback,
    };
    sapps_signal = App_SharedSignal_Create(
        0, App_AcceleratorPedalSignals_IsSappsAlarmActive, world,
        sapps_callback);
    App_FsmWorld_RegisterSignal(world, sapps_signal);

Reply:
Hmm, so I think I still prefer this in the `Create` function, because since we would presumably pass in the callback functions to the world function constructor, this forces any users of the world to properly setup the callback. I suppose I'm trying to minimize any duplication that might be required between here and the tests.

Reply:
That's fine with me too. I will make `RegisterSignal` private and also create another private function `UnregisterSignal` to clean things up in `World_Destroy`.